validphys2/src/validphys/commondataparser.py

```diff
@@ -1,18 +1,319 @@
+        if self.variants is None:
+            raise ValueError(f"There are no variants defined for {self.setname}")
+        if variant not in self.variants:
+            raise ValueError(f"The variant {variant} is not defined for {self.setname}")
+        # If there were not enabled variants, save the current state of the class
+        if self._enabled_variant is None:
+            # A shallow copy is enough because the variants update attributes of this class
+            self._default = copy(self.__dict__)
+            self._enabled_variant = []
+
+        self.__dict__.update(self.variants[variant].__dict__.items())
+        self._enabled_variant.append(variant)
+
+    def disable_variants(self):
+        """Get the CommonMetaData back to its original state"""
+        if self._default is not None:
+            self.__dict__.update(self._default)
+
+
+# TODO: will be moved to coredata.py and will substitute CommonData.py
+# and obviously the folder here is only for development purposes
+_folder_data = Path(__file__).parent / "../../../buildmaster"
+
+
+@dataclass
+class _CommonData:
+    """
+    Data, kinematics and uncertainties contained in Commondata files.
+    A CommonData is only defined by its name and the enabled variant.
+
+    The information from CommonData is provided by the following properties
+        - metadata: all metadata information for the dataset
+        - data: central data for the dataset
+        - uncertainties: uncertainties of the dataset
+        - kinematics: kinematic information
+    """
+
+    name: str
+    variant: Optional[str] = None
+
+    def enable_variant(self, new_variant):
+        if new_variant != self.variant:
+            self.variant = new_variant
+            self.metadata.enable_variant(self.variant)
+            # Delete loaded information as it will need to be recomputed
+            del self.data
+            del self.uncertainties
+
+    # The files where the CommonData information is retreived from as defined by the metadata
+    @property
+    def metadata_file(self):
+        return _folder_data / self.name / "metadata.yaml"
+
+    @property
+    def data_file(self):
+        return _folder_data / self.name / self.metadata.data_central
+
+    @property
+    def uncertainity_files(self):
+        return [_folder_data / self.name / i for i in self.metadata.data_uncertainties]
+
+    @property
+    def kinematics_file(self):
+        return _folder_data / self.name / self.metadata.kinematics.file
+
+    @property
+    def ndata(self):
+        return self.metadata.ndata
+
+    @cached_property
+    def metadata(self):
+        ret = parse_yaml_inp(self.metadata_file, CommonMetaData)
+        if self.variant:
+            ret.enable_variant(self.variant)
+        return ret
+
+    @cached_property
+    def data(self):
+        """Pandas DataFrame containing the central data"""
+        datayaml = yaml.safe_load(self.data_file.read_text(encoding="utf-8"))
```

Comment:
```suggestion
        datayaml = yaml.safe_load(self.data_file)
```

Reply:
I don't think this works?
```
from reportengine.compat import yaml
from pathlib import Path
yaml.safe_load(Path("run.yml"))
```
results in
```
YAMLStreamError: stream argument needs to have a read() method
```

Reply:
Right, that is sily. This does work:

```
In [6]: y = yaml.YAML()

In [7]: y.load(Path("xx.yml"))
```

Reply:
and also e.g.
```
In [12]: y.load("a: 1")
Out[12]: ordereddict([('a', 1)])
```

Reply:
I'll stay with `safe_load` since that seems to be what's used in most of vp.

Reply:
I guess one do `open` to get a stream, which avoids going over the file. In any case I think I'd be nice if `parse_yaml_inp` worked for whatever you feed it.